web/src/app/rotations/RotationForm.js

```diff
@@ -1,54 +1,94 @@
-import React from 'react'
+import React, { useMemo, useState } from 'react'
 import p from 'prop-types'
+import {
+  TextField,
+  Grid,
+  MenuItem,
+  Typography,
+  makeStyles,
+  Switch,
+  FormControlLabel,
+} from '@material-ui/core'
+import { startCase } from 'lodash'
+import { DateTime } from 'luxon'
+import { useQuery, gql } from '@apollo/client'
+
 import { FormContainer, FormField } from '../forms'
 import { TimeZoneSelect } from '../selection'
-import { TextField, Grid, MenuItem } from '@material-ui/core'
-import { startCase } from 'lodash'
-import { DateTime, Info } from 'luxon'
-import { ISOTimePicker } from '../util/ISOPickers'
+import { ISODateTimePicker } from '../util/ISOPickers'
+import NumberField from '../util/NumberField'
+import Spinner from '../loading/components/Spinner'
+
+const query = gql`
+  query nextRotationHandoffTimes($input: NextRotationHandoffTimesInput) {
+    nextRotationHandoffTimes(input: $input)
+  }
+`
 
 const rotationTypes = ['hourly', 'daily', 'weekly']
 
-export default function RotationForm(props) {
-  function dayOfWeek() {
-    const { start, timeZone } = props.value
-    return DateTime.fromISO(start, { zone: timeZone }).weekday
-  }
+const useStyles = makeStyles({
+  listNone: {
+    listStyle: 'none',
+  },
+  bolder: {
+    fontWeight: 'bolder',
+  },
+  flex: {
+    display: 'flex',
+  },
+  height7rem: {
+    height: '7rem',
+  },
+  noSpacing: {
+    margin: 0,
+    padding: 0,
+  },
```

Comment:
nit: when using classNames, I generally set the name relative to what I'm changing within the JSX, and let the CSS do the talking

For example (though you don't have necessarily to use these):

```suggestion
  handoffTimestamp: {
    listStyle: 'none',
  },
  handoffsTitle: {
    fontWeight: 'bolder',
  },
  tzContainer: {
    display: 'flex',
  },
  handoffsTitleContainer: {
    height: '7rem',
  },
  handoffsList: {
    margin: 0,
    padding: 0,
  },
```